test/spec/directives/bsSwitchSpec.js

```diff
@@ -674,18 +674,30 @@ describe('Directive: bsSwitch', function () {
 
   function makeTestChange(input) {
     return function () {
-      compileDirective('change', input);
+      var element = compileDirective('change', input);
 
       scope.change = function() {};
       spyOn(scope, 'change');
 
-      scope.model = false;
+      element.find('input').bootstrapSwitch('toggleState');
+      scope.$apply();
+      expect(scope.change).toHaveBeenCalled();
+      scope.change.calls.reset();
+
+      element.find('input').bootstrapSwitch('toggleState');
+      scope.$apply();
+      expect(scope.change).toHaveBeenCalled();
+      scope.change.calls.reset();
+
+      scope.model = undefined;
       scope.$apply();
       expect(scope.change).not.toHaveBeenCalled();
```

Comment:
Shouldn't this trigger a change?
Action that should trigger a change:

* model false --> model true
* model false --> model undefined/null
* model true --> model false
* model true --> model undefined/null
* model undefined/null --> model true
* model undefined/null --> model false
* model undefined/null --> view true
* model true --> view false
* model false --> view true

I'd split this test in two, one for the model changes and one for the view changes.